monai/transforms/signal/array.py

```diff
@@ -0,0 +1,773 @@
+        self.boundaries = boundaries
+        self.frequencies = frequencies
+        self.fraction = fraction
+
+    def _randomize(self):
+        super().randomize(None)
+        self.magnitude = self.R.uniform(low=self.boundaries[0], high=self.boundaries[1])
+        self.fracs = self.R.uniform(low=self.fraction[0], high=self.fraction[1])
+        self.freqs = self.R.uniform(low=self.frequencies[0], high=self.frequencies[1])
+
+    def _paste_slices(self, tup):
+        pos, w, max_w = tup
+        max_w = max_w.shape[len(max_w.shape) - 1]
+        wall_min = max(pos, 0)
+        wall_max = min(pos + w, max_w)
+        block_min = -min(pos, 0)
+        block_max = max_w - max(pos + w, max_w)
+        block_max = block_max if block_max != 0 else None
+        return slice(wall_min, wall_max), slice(block_min, block_max)
+
+    def _paste(self, wall, block, loc):
+        loc_zip = zip(loc, block.shape, wall)
+        wall_slices, block_slices = zip(*map(self._paste_slices, loc_zip))
+
+        wall[:, wall_slices[0]] = block[block_slices[0]]
+
+        if wall.shape[0] == 1:
+            wall = wall.squeeze()
+        return wall
+
+    def __call__(self, signal: np.ndarray) -> np.ndarray:
+        """
+        Args:
+            signal: input 1 dimension signal to which a partial square pulse will be added
+        """
+        self._randomize()
+        if len(signal.shape) > 1:
+            length = signal.shape[len(signal.shape) - 1]
+        else:
+            signal = signal[np.newaxis, :]
+            length = signal.shape[1]
+
+        time_partial = np.arange(0, round(self.fracs * length), 1)
+        squaredpulse_partial = self.magnitude * square(self.freqs * time_partial)
+
+        loc = np.random.choice(range(length))
+        signal = self._paste(signal, squaredpulse_partial, (loc,))
+
+        if len(signal.shape) == 1:
+            signal = np.expand_dims(signal, axis=0)
+
+        return signal
+
+
+class SignalNormalize(Transform):
+    """
+    Normalize signal between 0 and 1
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __call__(self, signal: np.ndarray) -> np.ndarray:
+        """
+        Args:
+            signal: signal to be normalized
+        """
+        if len(signal.shape) == 1:
+            signal = np.expand_dims(signal, axis=0)
+
+        signal = (signal - np.min(signal)) / (np.max(signal) - np.min(signal))
+        return signal
+
+
+class SignalStandardize(Transform):
+    """
+    Standardize a signal
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __call__(self, signal: np.ndarray) -> np.ndarray:
+        """
+        Args:
+            signal: signal to be standardized
+        """
+        if len(signal.shape) == 1:
+            signal = np.expand_dims(signal, axis=0)
+
+        signal = (signal - signal.mean()) / signal.std()
+        return signal
+
+
+class SignalZeroPad(Transform):
```

Comment:
Think the same could be achieved with `SpatialPad(spatial_size=2000, method="end", mode="constant")`. Admittedly, `SpatialPad` is an ill-fitting name here.

Reply:
removed